### ISSUE TITLE
ChoiceQuestion undefined index

### ISSUE THREAD

**Issue by funivan** [symfony/symfony#15343]
Try to use ChoiceQuestion. Failure in any case
State:
```
> php --version
> PHP 5.5.9-1ubuntu4.11 (cli) (built: Jul  2 2015 15:23:08) 
```
How to reproduce: 
create directory 
`mkdir /tmp/cons && cd  /tmp/cons`
install console component
`composer require "symfony/console:^2.7"
`
create `test.php` 
```php
<?php
  require_once 'vendor/autoload.php';

  use Symfony\Component\Console\Application;
  use Symfony\Component\Console\Command\Command;
  use Symfony\Component\Console\Input\InputInterface;
  use Symfony\Component\Console\Output\OutputInterface;
  use Symfony\Component\Console\Question\ChoiceQuestion;

  class ColorCommand extends Command {

    protected function configure() {
      $this->setName('color');

    }

    protected function execute(InputInterface $input, OutputInterface $output) {
      $helper = $this->getHelper('question');
      $question = new ChoiceQuestion(
        'Please select your favorite color (defaults to red)',
        array('red', 'blue', 'yellow')
      );
      $question->setMultiselect(true);
      $question->setErrorMessage('Color %s is invalid.');

      $color = $helper->ask($input, $output, $question);
      $output->writeln('You have just selected: ' . $color);

    }
  }


  $application = new Application();
  $application->add(new ColorCommand());
  $application->run();
```

Run `php test.php color`
Select any color.

My result:
`
Notice: Undefined index: yellow in /tmp/cons/vendor/symfony/console/Question/ChoiceQuestion.php on line 165
`
More backtrace:
http://i.imgur.com/8hsLIwe.png
http://i.imgur.com/UDyRsS4.png

**Comment by ifmi** [comment]
This will only work if your key will be same as value.

In this color calse the array you are providing for ChoiceQuestion needs to be:

['red' =>'red', 'blue' => 'blue', 'yellow' => 'yellow']

**Comment by nWidart** [comment]
I'm having the same issues here.

**Comment by xabbuh** [comment]
This looks like a duplicate of #15309 which was fixed by #15271.

**Comment by nWidart** [comment]
It's not fixed for me.

**Comment by funivan** [comment]
@ifmi yes. With string keys it works fine even if array contain one sting key ` array('red'=>'red', 'green', 'blue' ) ` it will be good to rewrite documentation or fix this bug ;)
@xabbuh ok I write next comments in #15309

**Comment by psynaptic** [comment]
I updated to 2.7.x-dev and the problem of using numeric keys is resolved for me.

```
$ php test.php color
Please select the site to build:
  [0] red
  [1] blue
  [2] yellow
 > 1
You have just selected: blue
```

**Comment by xabbuh** [comment]
@nWidart The fix was merged after the release of 2.7.2 and will be part of Symfony 2.7.3.

**Comment by nWidart** [comment]
Awesome, thank you for the heads up. :smile:
